python/pyarrow/tests/test_orc.py

```diff
@@ -626,11 +626,20 @@ def test_wrong_usage_orc_writer(tempdir):
 
 def test_orc_writer_with_null_arrays(tempdir):
     from pyarrow import orc
-    import pyarrow as pa
 
     path = str(tempdir / 'test.orc')
     a = pa.array([1, None, 3, None])
     b = pa.array([None, None, None, None])
     table = pa.table({"int64": a, "utf8": b})
     with pytest.raises(pa.ArrowNotImplementedError):
         orc.write_table(table, path)
+
+
+def test_orc_writer_with_arrays_with_unsupported_types():
```

Comment:
There's a test on the C++ side, I don't think it's worth checking it a second time here. These tests should probably focus on exercising the Python APIs, not reproduce the existing C++ tests.